tools-and-tests/tools/src/main/java/org/hiero/block/tools/capacity/NetworkCapacityClient.java

```diff
@@ -0,0 +1,237 @@
+        System.out.println("Recording folder: " + recordingFolder);
+
+        // No explicit start; metrics starts on first data seen.
+        BlockStreamPublishClient publishClient = createPublishClient();
+        streamBlocksFromFolder(publishClient);
+
+        metrics.reportFinal();
+    }
+
+    private BlockStreamPublishClient createPublishClient() {
+        Duration timeoutDuration = Duration.ofMillis(config.readTimeoutMillis());
+        Tls tls = Tls.builder().enabled(false).build();
+
+        PbjGrpcClientConfig grpcConfig =
+                new PbjGrpcClientConfig(timeoutDuration, tls, Optional.empty(), "application/grpc");
+
+        Http2ClientProtocolConfig http2Config = Http2ClientProtocolConfig.builder()
+                .flowControlBlockTimeout(Duration.ofMillis(config.flowControlTimeoutMillis()))
+                .initialWindowSize(config.initialWindowSize())
+                .maxFrameSize(config.maxFrameSize())
+                .maxHeaderListSize(config.maxHeaderListSize())
+                .ping(config.pingEnabled())
+                .pingTimeout(Duration.ofMillis(config.pingTimeoutMillis()))
+                .priorKnowledge(config.priorKnowledge())
+                .build();
+
+        GrpcClientProtocolConfig grpcProtocolConfig = GrpcClientProtocolConfig.builder()
+                .abortPollTimeExpired(false)
+                .pollWaitTime(timeoutDuration)
+                .build();
+
+        WebClient webClient = WebClient.builder()
+                .baseUri("http://" + serverAddress + ":" + serverPort)
+                .tls(tls)
+                .protocolConfigs(List.of(http2Config, grpcProtocolConfig))
+                .connectTimeout(timeoutDuration)
+                .build();
+
+        PbjGrpcClient pbjGrpcClient = new PbjGrpcClient(webClient, grpcConfig);
+        return new BlockStreamPublishClient(pbjGrpcClient);
+    }
+
+    private void streamBlocksFromFolder(BlockStreamPublishClient publishClient) throws IOException {
+        List<Path> blockFiles = Files.list(recordingFolder)
+                .filter(path -> path.toString().endsWith(".blk.gz"))
+                .sorted()
+                .collect(Collectors.toList());
+
+        if (blockFiles.isEmpty()) {
+            System.err.println("No block files found in " + recordingFolder);
+            return;
+        }
+
+        System.out.println("Found " + blockFiles.size() + " block files to stream");
+
+        CompletableFuture<Void> streamingComplete = new CompletableFuture<>();
+        ResponseHandler responseHandler = new ResponseHandler(streamingComplete);
+        Pipeline<PublishStreamRequest> requestPipeline = publishClient.publishBlockStream(responseHandler);
+
+        for (Path blockFile : blockFiles) {
+            try {
+                streamBlockFile(blockFile, requestPipeline);
+                // Optional pacing; consider removing for tight windows
+                Thread.sleep(10);
```

Comment:
An improvement for later or now if small would be add this as a config item in serverDefaultConfig to an engineer can dictate block delays